### DISCUSSION TITLE
Proposal to change output of TransformVectorMult node in VisualShader from vec3 to vec4

### DISCUSSION THREAD

**Discussion by americoperez49** [godotengine/godot-proposals#8375] [General Discussions]
When using the TransformVectorMult node in VisualShader,currently, the TransformVectorMult node outputs a vec3 instead of a vec4, causing the w component of the vector to get dropped.

![Example of truncation](https://github.com/godotengine/godot/assets/17209617/7b9fe37e-20c5-4151-af4c-15c1df9aabc2)

From mine and QbieShay's understanding, TransformVectorMult should output a vec4. This would be used to do the following calculation, for example, using the Visual Shader:
```
    float depth_tex = textureLod(depth_texture,SCREEN_UV,0.0).r;
    vec4 world_pos = INV_PROJECTION_MATRIX * vec4(SCREEN_UV*2.0-1.0,depth_tex,1.0);
    world_pos.xyz/=world_pos.w;
```

If this is not the intended use case of the TransformVectorMult node and it is , by design, intended to return a vec3. Then what would be the recourse for performing the calculation mentioned above using the Visual Shader?

**Comment by AThousandShips** [comment]
As stated on the issue, you need to fill this out as a proposal, not an issue, this is not filled out correctly as a proposal, not just copying over the issue report text

**Reply by americoperez49** [reply]
According to the instructions:

"If you have a more general idea for a feature but are not well versed in Godot's architecture or do not possess the necessary knowledge to implement it in the engine, feel free to open a [discussion](https://github.com/godotengine/godot-proposals/discussions/new) instead of an [issue](https://github.com/godotengine/godot-proposals/issues/new/choose)."

Also:

"Note: The following points describe requirements for a proposal issue. A [discussion](https://github.com/godotengine/godot-proposals/discussions/new), on the other hand, can be started in any form."

I clicked on discussion and it sent me to this:
![image](https://github.com/godotengine/godot-proposals/assets/17209617/4969acc4-852b-4247-ba9b-6448bb1b70c8)


and then i clicked on General Discussion and it shows me this:
![image](https://github.com/godotengine/godot-proposals/assets/17209617/74d330ee-6bba-4dae-888b-71dc0c843e3d)


I'm not sure how what I am doing is incorrect since accordint to the instructions " A [discussion](https://github.com/godotengine/godot-proposals/discussions/new), on the other hand, can be started in any form."

**Reply by AThousandShips** [reply]
Yes, but you have simply copied your issue report, instead of writing a post describing what you *want* as a feature, this post is just confusing, and it *isn't* a proposal but an issue report, that's the issue here

You could argue by the same logic that posting a poem would be in line with:
> Note: The following points describe requirements for a proposal issue. A [discussion](https://github.com/godotengine/godot-proposals/discussions/new), on the other hand, can be started in any form.

But you agree it isn't really a valid proposal, right?

**Reply by AThousandShips** [reply]
Do you see how the wording and information in this discussion isn't relevant to this? That it should just be "I want XYZ", this is very hard to get what you are suggesting

**Reply by americoperez49** [reply]
I am not trying to say I want a certain feature. I am trying to have an open disscusion about the TransformVectorMult node, its current state, and its intended use with people that are well versed in the engines architecture and code base, but I get your point . Let me rewrite the post as if I was proposing a change.

**Reply by americoperez49** [reply]
@AThousandShips , sorry for the bother but would you mind double checking that my post is ok now? I really dont want to be a nuisance here and I actually do want to try to help the best I can. So I want to make sure I'm abiding by the rules around here.

**Reply by AThousandShips** [reply]
Looks great, thank you for putting up with me, it's just so that everyone can see what is wanted and so that what you need for your projects can be realized :slightly_smiling_face:

**Comment by americoperez49** [comment]
Turns out there is a dedicated node called proximity_fade that you can use to do this calculation. 

This discussion can be closed.
